Client/Frontend/Home/FirefoxHomeViewController.swift

```diff
@@ -207,6 +207,32 @@ class FirefoxHomeViewController: UICollectionViewController, HomePanel, GleanPlu
         currentTab?.lastKnownUrl?.absoluteString.hasPrefix("internal://") ?? false ? BrowserViewController.foregroundBVC().urlBar.leaveOverlayMode() : nil
     }
 
+    func updatePocketCellsWithVisibleRatio(cells: [UICollectionViewCell], relativeRect: CGRect) {
+        guard let window = UIApplication.shared.window else { return }
+        for cell in cells {
+            // For every story cell get it's frame realtive to the window
+            let targetRect = cell.superview.map { window.convert(cell.frame, from: $0) } ?? .zero
+
+            let ratio = targetRect.visibilityRatio(relativeTo: relativeRect)
```

Comment:
What are we doing with this ratio?

Reply:
As I understood from the Android team, one condition to mark the sponsored story as "seen by the user" is to check if the card was shown on screen at least 50% or higher. This code checks that ratio but it's not yet used.

Reply:
Ah ok I see, so it will be used in a upcoming story. Since the `ratio` variable isn't used it will generate a warning. So for now could you fix the warning with `let _ = [..]` and write a comment explaining the context (with upcoming story for the TODO) so any devs playing in that area will be aware of this too 🙏

Reply:
Updated